suture_simple_test.go

```diff
@@ -5,12 +5,12 @@ import "fmt"
 type Incrementor struct {
 	current int
 	next    chan int
-	stop    chan bool
+	stop    chan struct{}
 }
 
 func (i *Incrementor) Stop() {
 	fmt.Println("Stopping the service")
-	i.stop <- true
+	close(i.stop)
```

Comment:
This is an important change, as there can be two execution paths:

1. What you anticipated when writing the example:
    1. `Stop` is called, `true` is send
    2. `true` is received in the `select` of `Serve`
    3. `Serve` sends another `true`
    4. `true` is received in the "`main`" to terminate after the log has happened
2. What is sub-optimal for demo purposes:
    1. `Stop` is called, `true` is send
    2. `true` is received in the "`main`" to terminate after the log has happened
    3. the example terminates, while never actually stopping the `Serve` go routine which continues to run forever

By closing the channel instead, both receive statements will immediately receive when reached, and therefore the go routine is always terminated. I think the example should not have this kind of bug, but implement a proper stopping mechanism to prevent this bug in implementations.